aurorawatch-pipeline/test_aurora_extract.py

```diff
@@ -0,0 +1,103 @@
+'''Test file for the AuroraWatch extract script.'''
+# pylint: disable=W0212
+
+from unittest.mock import patch
+import pytest
+
+from requests import Response
+
+from aurora_extract import make_request, get_status, extract
+
+
+class TestMakeRequest():
+    '''Tests for the make request function.'''
+
+    @patch('aurora_extract.requests.get')
+    def test_make_request_success_returns_bytes(self, mock_get, valid_xml):
+        '''Tests that the a bytes object is returned when the
+        request is successful.'''
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_response._content = valid_xml
+        mock_get.return_value = mock_response
+
+        data = make_request()
+
+        assert mock_get.called
+        assert data == valid_xml
+
+    @patch('aurora_extract.requests.get')
+    def test_make_request_fail(self, mock_get):
+        '''Tests that an error is raised if the request fails.'''
+        mock_response = Response()
+        mock_response.status_code = 404
+        mock_get.return_value = mock_response
+
+        with pytest.raises(ConnectionError):
+            make_request()
+
+
+class TestGetStatus():
+    '''Tests for the get status function.'''
+
+    def test_get_status_success(self, valid_xml):
+        '''Tests that get status runs correctly and returns a string
+        given a valid xml as bytes input.'''
+        result = get_status(valid_xml)
+
+        assert result == 'green'
+
+    @pytest.mark.parametrize('xml_input', [(1), (['list', 'test']),
```

Comment:
Valid use of parameterisation, reducing redundancy in your testing suite and allows you to cover a wide range of edge-cases while keeping your code readable. (Alex).